pytorch_lightning/metrics/functional/classification.py

```diff
@@ -1018,11 +1019,19 @@ def iou(
         target: torch.Tensor,
         ignore_index: Optional[int] = None,
         absent_score: float = 0.0,
+        threshold: float = 0.5,
         num_classes: Optional[int] = None,
         reduction: str = 'elementwise_mean',
 ) -> torch.Tensor:
     """
-    Intersection over union, or Jaccard index calculation.
+    Intersection over union, or Jaccard index calculation. Works with binary, multiclass, and multilabel data.
```

Comment:
please move the functional implementation to its own `iou.py` file (still in the functional folder). As functions gets updated and unified the are being removed from the `classification.py` file.

Reply:
How about iou test file? Do you want to add a separate file?

Reply:
Yes add seperate file in `tests/metrics/classification`. I recommend structuring it in a similar fashion as `test_confusion_matrix.py`